### ISSUE TITLE
Gracefully recover after printer goes offline

### ISSUE THREAD

**Issue by mtehonica** [t0nyz0/BambuBoard#7]
When the printer is rebooted and becomes unreachable, the service doesn't appear to recover on its own. It requires restarting the application to recover.

```
Connection error: Error: read ECONNRESET
Connection error: Error [ERR_STREAM_DESTROYED]: Cannot call write after a stream was destroyed
```

**Comment by t0nyz0** [comment]
I've noticed this myself. I'll see if I can provide a fix in the next version.

**Comment by t0nyz0** [comment]
I just released v1.0.8. In this version I've rebuilt the connection/disconnection/error retry logic. After my limited testing everything seems to be much more stable. The dashboard automatically comes back online once the printer is available. 

Let me know if you run into any issues.
